kong/runloop/handler.lua

```diff
@@ -708,13 +708,18 @@ do
   end
 
 
-  get_router = function()
-    return router
+  rebuild_router = function(timeout)
+    return rebuild("router", update_router, router_version, router_semaphore, timeout)
   end
 
 
-  rebuild_router = function(timeout)
-    return rebuild("router", update_router, router_version, router_semaphore, timeout)
+  get_updated_router = function()
+    local ok, err = rebuild_router(REBUILD_TIMEOUT)
+    if not ok then
+      log(ERR, "no router to route request (reason: ", err, ")")
```

Comment:
There's a difference here between `get_updated_plugins_iterator` and this new `get_updated_router` and : the former logs the error *and returns the old plugins iterator*, while this one cancels the request and returns a 500 error right away. I am not sure which one is the best approach: erroring is more "correct" while returning the old one is "more resilient".

Reply:
I would expect consistency on both. I have no hard opinion on this, but would lean toward resiliency (the new build is tried again).

Reply:
After reading #4194 and discussing this privately with you, I decided to make the change so that both behave in the same way (logging and returning old instead of failing the request)

I have also decided to move that change to #4627 instead of keeping it here, because it looks non-controversial enough to me.